src/OpenDiffix.Core/Expression.fs

```diff
@@ -215,6 +216,12 @@ let rec evaluateScalarFunction fn args =
   | Cast, [ Real r; String "text" ] -> r.ToString(doubleStyle) |> String
   | Cast, [ Boolean b; String "text" ] -> b.ToString().ToLower() |> String
 
+  | NullIf, [ Boolean x; Boolean y ] -> if x = y then Null else Boolean x
+  | NullIf, [ Integer x; Integer y ] -> if x = y then Null else Integer x
+  | NullIf, [ Real x; Real y ] -> if x = y then Null else Real x
+  | NullIf, [ String x; String y ] -> if x = y then Null else String x
+  | NullIf, [ Null; Null ] -> Null
```

Comment:
Can't we just generalize to 2 cases:

```fs
NullIf, [ x, y ] when x = y -> Null
NullIf, [ x, _y ] -> x
```

Maybe even the Exliir way works: `NullIf, [ x; x ] -> Null` (not sure if F# supports it)

The F# `Value` ADT will have an (by value) equality operator out of the box so no need to worry about deep comparison.

Reply:
Heh, Elixir way doesn't work, tried that by muscle memory first.

But the 2-cases way doesn't break on disparate types, which PostgreSQL does break on. If we don't need the type check and have `nullif(1, 'a')` be just `1`, we can do that. 

Hm, on second thought, PostgreSQL has no problem with `nullif(1.1, 1)`, doing a cast somewhere there. However, in our case, the shallow `=` of the 2 case solution will still not do that trick, we would need to have a bunch of special cases (for example, PostgreSQL's `nullif(1.0, 1)` produces NULL). Hard for me to judge.

Reply:
We can have a general `Value.equivalent` or `Value.softEquals` function which handles the real and integer comparison. But in any case, for `count == 0` we have clearly defined types and we don't care about SQL/Postgres compliance.

Reply:
I say strict equality is fine here considering your current code already does that.

Reply:
kk, I'll simplify then

Reply:
lol, I've been scratching my head for a while, why would our simplified `nullif(Integer 1, Real 1.0)` resolve to Null. It turns out there's a generic coercion for binary functions matching before the `NullIf` case:

```
// Treat mixed integer/real binary operations as real/real operations.
| _, [ Integer i; Real r ] -> evaluateScalarFunction fn [ Real(double i); Real r ]
| _, [ Real r; Integer i ] -> evaluateScalarFunction fn [ Real r; Real(double i) ]
```

I guess we can keep this behavior, this is in line with PostgreSQL and common sense.

Reply:
Hah, I haven't been aware of this either!

> I guess we can keep this behavior, this is in line with PostgreSQL and common sense.

Yes, looks good.